### PR TITLE
Expose VM properties to the user

### PR DESCRIPTION
**Pull request by kayousterhout**
The point of this PR is to expose information about the memory and CPUs on each worker so that blueprints can use it (see https://github.com/kelda/spark/pull/22 for an example).  There's one other commit here to fix an existing problem that I uncovered in the process of adding this (the other issue needed to be fixed to make the unit tests with the new change).

### CODE REVIEW

File: js/bindings/bindings.js [hunk 1]
```diff
@@ -700,12 +700,12 @@ class Machine {
    *   size: '512mb',
    * });
    *
-   * @param {Object.<string, string>} [opts] - Optional arguments that
-   *   modify the machine.
-   * @param {string} [opts.provider] - The cloud provider that the machine
+   * @param {Object.<string, string>} opts - Arguments that
```
Comment (kklin) [inline js/bindings/bindings.js]: Nit - could "modify" be on the same line as "that"? The line break looks a little funny

File: js/bindings/bindings.js [hunk 2]
```diff
@@ -866,15 +881,15 @@ class Machine {
         continue;
       }
 
-      if (bestSize === '' || description.Price < bestPrice) {
-        bestSize = description.Size;
+      if (bestMachine === undefined || description.Price < bestPrice) {
+        bestMachine = description;
         bestPrice = description.Price;
```
Comment (kklin) [inline js/bindings/bindings.js]: I have a small preference to get rid of the `bestPrice` variable and reference `bestMachine.Price`, what do you think?

File: js/bindings/bindings.js [hunk 3]
```diff
@@ -114,8 +114,13 @@ class Infrastructure {
       containers.push(c.toKeldaRepresentation());
     });
 
+    const machines = [];
+    this.machines.forEach((m) => {
```
Comment (kklin) [inline js/bindings/bindings.js]: Might be cleaner to express this as `this.machines.map`? Don't feel strongly though

Comment (kayousterhout) [reply js/bindings/bindings.js]: Oh nice I didn't realize JS had a map!

File: js/bindings/bindings.js (hunk 2 above)
Comment (kayousterhout) [reply js/bindings/bindings.js]: Oh that's much nicer - thanks!

**Pull request merged**